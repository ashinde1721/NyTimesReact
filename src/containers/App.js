import React from 'react';
import './App.scss';
import Layout from '../components/Layout/Layout';
import Articles from './Articles/Articles';
import Details from './Details/Details';
import Error404 from '../components/Error404/index';
import { Route, Switch, Redirect, BrowserRouter as Router, } from 'react-router-dom';

function App() {
  return (
    <div>
      <Router>
        <Layout>
          <Switch>
            <Route exact path="/" component={Articles} />
            <Route path="/details" component={Details} />
            <Route path="/error404" component={Error404} />
            <Redirect from="*" to="/error404" />
          </Switch>
        </Layout>
      </Router>
    </div>
  );
}

export default App;
