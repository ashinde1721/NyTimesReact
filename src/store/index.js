import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducer";
import createSagaMiddleware from 'redux-saga';

import {getArticleSaga} from  '../sagas/articlesSaga'

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(getArticleSaga);

export default store;