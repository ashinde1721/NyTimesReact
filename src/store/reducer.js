import * as actions from '../actions/actions';

const initialState = {
    search: '',
    articlesList: []
};

const reducer = (state = initialState, action) => {
    const { type, payload = {} } = action;

    switch (type) {
        case actions.SEARCH_TEXT:
            return { ...state, search: payload.search };

        case actions.SAVE_ARTICLES:
            return {
                ...state,
                articlesList: payload.articlesList
            }
        default:
            return state;
    }
};

export default reducer;
