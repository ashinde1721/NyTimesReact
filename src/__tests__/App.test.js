import React from 'react';
import { Provider } from 'react-redux';
import { cleanup, render } from 'react-testing-library';
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';

import store from '../store/index'
import App from '../containers/App';

describe('App Component', () => {
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  afterEach(() => cleanup());

  it('renders component and match ui snapshot to cover UI changes', () => {
    expect(renderedComponent.container).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  })
});
