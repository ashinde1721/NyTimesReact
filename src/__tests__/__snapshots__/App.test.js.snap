// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders component and match ui snapshot to cover UI changes 1`] = `
<div>
  <div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <nav
          class="navbar aqua-marine"
        >
          <div
            class="navbar-brand white"
          >
            <div
              class="row"
            >
              <div
                class="col-6"
              >
                <img
                  alt=""
                  class="align"
                  height="20"
                  src="white-menu.jpg"
                  width="20"
                />
                <span
                  class="pad-left"
                >
                  NY Times Most Popular
                </span>
              </div>
              <div
                class="col-6"
              >
                <div
                  class="float-right"
                >
                  <div>
                    <img
                      alt=""
                      class="align"
                      height="20"
                      src="search.png"
                      width="20"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </nav>
        <main
          class="content"
        >
          <div
            class="row"
          >
            <div
              class="col-md-12"
            />
          </div>
        </main>
      </div>
    </div>
  </div>
</div>
`;
